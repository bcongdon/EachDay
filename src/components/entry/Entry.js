import React, { Component } from 'react'
import { Grid, Rating, Dropdown, Statistic } from 'semantic-ui-react'
import { PropTypes } from 'prop-types'
import moment from 'moment'
import { openEntryModal, deleteEntry } from '../../actions'
import { connect } from 'react-redux'

class Entry extends Component {
  static propTypes = {
    rating: PropTypes.number,
    notes: PropTypes.string,
    date: PropTypes.string.isRequired,
    openEntryModal: PropTypes.func.isRequired,
    id: PropTypes.number.isRequired,
    deleteEntry: PropTypes.func
  }

  constructor (props) {
    super(props)
    this.onEditClick = this.onEditClick.bind(this)
    this.onDeleteClick = this.onDeleteClick.bind(this)
  }

  onEditClick () {
    this.props.openEntryModal({
      rating: this.props.rating,
      notes: this.props.notes,
      date: this.props.date,
      id: this.props.id
    })
  }

  onDeleteClick () {
    this.props.deleteEntry(this.props.id)
  }

  render () {
    return (
      <Grid.Row>
        <Grid.Column width={2} textAlign='center' style={{minWidth: 100, margin: '0 auto'}}>
          <Statistic size='small'>
            <Statistic.Label>{moment(this.props.date).format('MMM')}</Statistic.Label>
            <Statistic.Value>{moment(this.props.date).format('DD')}</Statistic.Value>
          </Statistic>
          <div>
            {this.props.rating ? <Rating style={{maxWidth: '100%', marginBottom: 20}} disabled size='tiny' maxRating={5} rating={this.props.rating} /> : null}
          </div>
        </Grid.Column>
        <Grid.Column textAlign='left' computer={10} tablet={10} mobile={16}>
          {this.props.notes}
        </Grid.Column>
        <Grid.Column width={1} verticalAlign='bottom' floated='right' style={{marginBottom: 0}}>
          <Dropdown icon='setting' className={'top right'}>
            <Dropdown.Menu style={{right: 0, left: 'auto'}}>
              <Dropdown.Item onClick={this.onEditClick} icon='edit' content='Edit' />
              <Dropdown.Item onClick={this.onDeleteClick} icon='delete' content='Delete' />
            </Dropdown.Menu>
          </Dropdown>
        </Grid.Column>
      </Grid.Row>
    )
  }
}

export default connect(null, { openEntryModal, deleteEntry })(Entry)
