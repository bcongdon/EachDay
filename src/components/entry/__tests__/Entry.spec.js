import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Entry } from '../Entry'
import { Dropdown } from 'semantic-ui-react'

jest.mock('universal-cookie')

const openEntryModal = jest.fn()
const deleteEntry = jest.fn()

describe('entry component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should render a basic entry correctly', () => {
    var dash = shallow(
      <Entry
        id={1}
        rating={5}
        date='2017-01-01'
        notes='Foo bar baz'
        openEntryModal={openEntryModal}
        deleteEntry={deleteEntry}
        />
    )
    expect(toJson(dash)).toMatchSnapshot()
  })

  it('should render without notes', () => {
    var dash = shallow(
      <Entry
        id={1}
        rating={5}
        date='2017-01-01'
        openEntryModal={openEntryModal}
        deleteEntry={deleteEntry}
        />
    )
    expect(toJson(dash)).toMatchSnapshot()
  })

  it('should render without a rating', () => {
    var dash = shallow(
      <Entry
        id={1}
        date='2017-01-01'
        notes='Foo bar baz'
        openEntryModal={openEntryModal}
        deleteEntry={deleteEntry}
        />
    )
    expect(toJson(dash)).toMatchSnapshot()
  })

  it('should call openEntryModel on edit click', () => {
    var dash = mount(
      <Entry
        id={1}
        date='2017-01-01'
        notes='Foo bar baz'
        openEntryModal={openEntryModal}
        deleteEntry={deleteEntry}
        />
    )
    dash.find(Dropdown.Item).findWhere((n) => n.props().icon === 'delete').simulate('click')
    expect(deleteEntry).toBeCalled()
  })

  it('should call deleteEntry on delete click', () => {
    var dash = mount(
      <Entry
        id={1}
        date='2017-01-01'
        notes='Foo bar baz'
        openEntryModal={openEntryModal}
        deleteEntry={deleteEntry}
        />
    )
    dash.find(Dropdown.Item).findWhere((n) => n.props().icon === 'edit').simulate('click')
    expect(openEntryModal).toBeCalled()
  })
})
