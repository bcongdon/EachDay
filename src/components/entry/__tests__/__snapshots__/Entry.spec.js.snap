// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entry component should render a basic entry correctly 1`] = `
<GridRow>
  <GridColumn
    style={
      Object {
        "margin": "0 auto",
        "minWidth": 100,
      }
    }
    textAlign="center"
    width={2}
  >
    <_
      name="entry-1"
    />
    <Statistic
      size="small"
    >
      <StatisticLabel>
        Jan
      </StatisticLabel>
      <StatisticValue>
        01
      </StatisticValue>
    </Statistic>
    <div>
      <Rating
        clearable="auto"
        disabled={true}
        maxRating={5}
        rating={5}
        size="tiny"
        style={
          Object {
            "marginBottom": 20,
            "maxWidth": "100%",
          }
        }
      />
    </div>
  </GridColumn>
  <GridColumn
    computer={10}
    mobile={16}
    tablet={10}
    textAlign="left"
  >
    Foo bar baz
  </GridColumn>
  <GridColumn
    floated="right"
    style={
      Object {
        "marginBottom": 0,
      }
    }
    verticalAlign="bottom"
    width={1}
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      className="top right"
      closeOnBlur={true}
      icon="setting"
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      selectOnBlur={true}
    >
      <DropdownMenu
        style={
          Object {
            "left": "auto",
            "right": 0,
          }
        }
      >
        <DropdownItem
          content="Edit"
          icon="edit"
          onClick={[Function]}
        />
        <DropdownItem
          content="Delete"
          icon="delete"
          onClick={[Function]}
        />
      </DropdownMenu>
    </Dropdown>
  </GridColumn>
</GridRow>
`;

exports[`entry component should render without a rating 1`] = `
<GridRow>
  <GridColumn
    style={
      Object {
        "margin": "0 auto",
        "minWidth": 100,
      }
    }
    textAlign="center"
    width={2}
  >
    <_
      name="entry-1"
    />
    <Statistic
      size="small"
    >
      <StatisticLabel>
        Jan
      </StatisticLabel>
      <StatisticValue>
        01
      </StatisticValue>
    </Statistic>
    <div />
  </GridColumn>
  <GridColumn
    computer={10}
    mobile={16}
    tablet={10}
    textAlign="left"
  >
    Foo bar baz
  </GridColumn>
  <GridColumn
    floated="right"
    style={
      Object {
        "marginBottom": 0,
      }
    }
    verticalAlign="bottom"
    width={1}
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      className="top right"
      closeOnBlur={true}
      icon="setting"
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      selectOnBlur={true}
    >
      <DropdownMenu
        style={
          Object {
            "left": "auto",
            "right": 0,
          }
        }
      >
        <DropdownItem
          content="Edit"
          icon="edit"
          onClick={[Function]}
        />
        <DropdownItem
          content="Delete"
          icon="delete"
          onClick={[Function]}
        />
      </DropdownMenu>
    </Dropdown>
  </GridColumn>
</GridRow>
`;

exports[`entry component should render without notes 1`] = `
<GridRow>
  <GridColumn
    style={
      Object {
        "margin": "0 auto",
        "minWidth": 100,
      }
    }
    textAlign="center"
    width={2}
  >
    <_
      name="entry-1"
    />
    <Statistic
      size="small"
    >
      <StatisticLabel>
        Jan
      </StatisticLabel>
      <StatisticValue>
        01
      </StatisticValue>
    </Statistic>
    <div>
      <Rating
        clearable="auto"
        disabled={true}
        maxRating={5}
        rating={5}
        size="tiny"
        style={
          Object {
            "marginBottom": 20,
            "maxWidth": "100%",
          }
        }
      />
    </div>
  </GridColumn>
  <GridColumn
    computer={10}
    mobile={16}
    tablet={10}
    textAlign="left"
  />
  <GridColumn
    floated="right"
    style={
      Object {
        "marginBottom": 0,
      }
    }
    verticalAlign="bottom"
    width={1}
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      className="top right"
      closeOnBlur={true}
      icon="setting"
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      selectOnBlur={true}
    >
      <DropdownMenu
        style={
          Object {
            "left": "auto",
            "right": 0,
          }
        }
      >
        <DropdownItem
          content="Edit"
          icon="edit"
          onClick={[Function]}
        />
        <DropdownItem
          content="Delete"
          icon="delete"
          onClick={[Function]}
        />
      </DropdownMenu>
    </Dropdown>
  </GridColumn>
</GridRow>
`;
